### PR TITLE
Feature/9 iOS edit text

### PR DESCRIPTION
**Pull request by Yarin18**

### CODE REVIEW

File: iosApp/iosApp/ui/textFields.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                Text(topLabel)
+            }
+            if (!isSecure) {
+                TextField("Type here", text: $input)
+                    .padding()
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 10)
+                            .stroke(Color.blackPearl, lineWidth: 1)
+                    )
+                if let errorMessage = errorMessage {
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/textFields.swift]: Try extracting this, check the DRY-principle

File: iosApp/iosApp/ui/textFields.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import SwiftUI
+
+struct EditText: View {
+    @Binding var input: String
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/textFields.swift:4]: Could you explain why u used @Binding and not @State and what the difference is?

File: iosApp/iosApp/ui/textFields.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            .overlay(
+                Button(action: {
+                    isPasswordVisible.toggle()
+                }) {
+                    Image(systemName: isPasswordVisible ? "eye.slash.fill" : "eye.fill")
+                        .foregroundColor(.blackPearl)
+                        .padding(.top, 30)
+                    
+                }
+                    .padding(.trailing, 8)
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/ui/textFields.swift]: I find it very confusing on what these paddings etc are, is there a way to improve this readability?

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: iosApp/iosApp/ui/textFields.swift (hunk 2 above)
Comment (Yarin18) [reply iosApp/iosApp/ui/textFields.swift:4]: @Binding creates a two-way connection between view & property, while @State is a single source of truth. From what I could tell @State is usually used in view models only

File: iosApp/iosApp/ui/textFields.swift (hunk 3 above)
Comment (Yarin18) [reply iosApp/iosApp/ui/textFields.swift]: Is my latest commit better in your opinion? The only downside is that you will always need to specify each "padding direction", so you'll always need to pass in 4 parameters

**Review by DanneD-Wisemen — APPROVED** [review]

**Pull request merged**